.bazelci/presubmit.yml

```diff
@@ -3,60 +3,148 @@ buildifier:
-      - "-//gazelle/..."
-      # As a regression test for #225, check that wheel targets still build when
-      # their package path is qualified with the repo name.
-      - "@rules_python//examples/wheel/..."
-      # We control Bazel version in integration tests, so we don't need USE_BAZEL_VERSION for tests.
-    skip_use_bazel_version_for_test: true
-    test_targets:
-      - "--" # Allows negative patterns; hack for https://github.com/bazelbuild/continuous-integration/pull/245
-      - "..."
-      # Gazelle is not fully Windows compatible: https://github.com/bazelbuild/bazel-gazelle/issues/1122
-      - "-//gazelle/..."
-      # The dependencies needed for this test are not cross-platform: https://github.com/bazelbuild/rules_python/issues/260
-      - "-//tests:pip_repository_entry_points_example"
+    <<: *reusable_config
+    name: Default test on Windows
+    platform: windows
+    test_flags:
+      - "--test_tag_filters=-integration-test,-fix-windows"
+  rbe:
+    <<: *reusable_config
+    name: Test on RBE
+    platform: rbe_ubuntu1604
     test_flags:
-      - "--test_tag_filters=-fix-windows"
-  rbe_ubuntu1604:
-    build_targets:
-      - "--" # Allows negative patterns; hack for https://github.com/bazelbuild/continuous-integration/pull/245
-      - "..."
-    # We control Bazel version in integration tests, so we don't need USE_BAZEL_VERSION for tests.
-    skip_use_bazel_version_for_test: true
-    test_targets:
-      - "--" # Allows negative patterns; hack for https://github.com/bazelbuild/continuous-integration/pull/245
-      - "..."
-      # TODO: The toolchain tests do not currently work in RBE
-      - "-//python/tests/toolchains/..."
-      # TODO: The integration tests do not currently work on RBE
-      # This list is the result of `bazel query 'filter(".*_example$", attr(generator_function, bazel_integration_test, //...))'`
-      - "-//examples:bzlmod_example"
-      - "-//examples:pip_install_example"
-      - "-//examples:pip_parse_example"
-      - "-//examples:pip_parse_vendored_example"
-      - "-//examples:pip_repository_annotations_example"
-      - "-//tests:pip_repository_entry_points_example"
+      - "--test_tag_filters=-integration-test,-acceptance-test"
+
+  integration_test_build_file_generation_linux:
+    <<: *reusable_build_test_all
+    name: build_file_generation integration tests on Linux
+    working_directory: examples/build_file_generation
+    platform: ubuntu2004
+  integration_test_build_file_generation_macos:
+    <<: *reusable_build_test_all
+    name: build_file_generation integration tests on macOS
+    working_directory: examples/build_file_generation
+    platform: macos
+  integration_test_build_file_generation_windows:
+    <<: *reusable_build_test_all
+    name: build_file_generation integration tests on Windows
+    working_directory: examples/build_file_generation
+    platform: windows
+
+  integration_test_bzlmod_linux:
```

Comment:
could you use https://github.com/bazelbuild/continuous-integration/blob/5a1a944d644bac905da8cab4111b14b6497b7316/buildkite/README.md#matrix-testing here to reduce the repetition?

Reply:
I didn't use it because I knew some wouldn't work on Windows. I fixed most of them, but there's still one test that needs fixing, then we can move to a matrix.